tests/filecheck/vector_ops.xdsl

```diff
@@ -0,0 +1,23 @@
+// RUN: xdsl-opt %s | xdsl-opt | filecheck %s
+
+builtin.module() {
+
+func.func() ["sym_name" = "vector_test", "function_type" = !fun<[], []>, "sym_visibility" = "private"] {
```

Comment:
I prefer having very modular tests (though that certainly isn't the standard in the current test suite). Therefore, I would suggest to change this into a function that takes and returns a memref such that you can remove `Alloca`, `Dealloc` and `Constant`. In this way, you will not rely on the `arith`and `memref` dialects as much and there will be less conflicts if we decide to change something there.